File: backend/flask/app/services/jwt_service.py
import jwt
import datetime
from config.settings import Config

class JwtService:
    @staticmethod
    def generate_token(email):
        payload = {
            'email': email,
            'iat': datetime.datetime.utcnow(),
            'exp': datetime.datetime.utcnow() + datetime.timedelta(seconds=Config.JWT_EXPIRATION),
        }
        return jwt.encode(payload, Config.JWT_SECRET, algorithm='HS256')

    @staticmethod
    def validate_token(token):
        try:
            jwt.decode(token, Config.JWT_SECRET, algorithms=['HS256'])
            return True
        except jwt.ExpiredSignatureError:
            return False
        except jwt.InvalidTokenError:
            return False

    @staticmethod
    def get_email_from_token(token):
        decoded = jwt.decode(token, Config.JWT_SECRET, algorithms=['HS256'])
        return decoded['email']

File: backend/laravel/app/Services/JwtService.php
<?php

namespace App\Services;

use Firebase\JWT\JWT;
use Firebase\JWT\Key;

class JwtService
{
    private $secretKey;
    private $expiration;

    public function __construct()
    {
        $this->secretKey = env('JWT_SECRET');
        $this->expiration = env('JWT_EXPIRATION');
    }

    public function generateToken($email)
    {
        $payload = [
            'email' => $email,
            'iat' => time(),
            'exp' => time() + $this->expiration,
        ];

        return JWT::encode($payload, $this->secretKey, 'HS256');
    }

    public function validateToken($token)
    {
        try {
            JWT::decode($token, new Key($this->secretKey, 'HS256'));
            return true;
        } catch (\Exception $e) {
            return false;
        }
    }

    public function getEmailFromToken($token)
    {
        $decoded = JWT::decode($token, new Key($this->secretKey, 'HS256'));
        return $decoded->email;
    }
}
